parambharat/ParlAI
# feature: add model and dictionary artifact logging

**Patch description**
This PR adds functionality to log model artifacts to Weights & Biases.

**Testing steps**
run the demo script with the wandb `args`
```
parlai train_model -t personachat -m transformer/ranker -mf /tmp/model_tr6 --n-layers 1 --embedding-size 300 --ffn-size 600 --n-heads 4 --num-epochs 2 -veps 0.25 -bs 64 -lr 0.001 --dropout 0.1 --embedding-type fasttext_cc --candidates batch --wandb-log True --wandb-project parlai --wandb-model True
```
The model is stored on wandb along with the logs for the run:
See [here](https://wandb.ai/parambharat/parlai/artifacts/model/model/v0/usage) for an example stored artifact and usage.
